services/auth/source/oauth2/store.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package oauth2
+
+import (
+	"fmt"
+	"net/http"
+
+	chiSession "gitea.com/go-chi/session"
+	"github.com/gorilla/sessions"
+)
+
+// SessionsStore creates a gothic store from our session
+type SessionsStore struct {
+}
+
+// Get should return a cached session.
+func (st *SessionsStore) Get(r *http.Request, name string) (*sessions.Session, error) {
+	chiStore := chiSession.GetSession(r)
+
+	rawData := chiStore.Get(name)
+	if rawData == nil {
+		return st.New(r, name)
+	}
+
+	oldSession, ok := rawData.(*sessions.Session)
+	if !ok {
+		return nil, fmt.Errorf("unexpected object in session: %v at name: %s", rawData, name)
```

Comment:
Maybe we should only print rawData to log.

Otherwise, if the rawData is returned with error string, and the error is shown to users, will it leak some sensitive data?

Reply:
Trace the calls back and you'll find that this error is never shown to the user (unless the server is in non-prod mode.)

The error should also never happen. If it does there has been a programming error in our code and we're overloading the oauth2 session key.

Reply:
I agree the error should never happen, but saving potential sensitive data to error string is not a good idea in my mind. Just like we shouldn't save user cookie or password to error string.

Reply:
IT IS NEVER PRINTED OUT.

Reply:
I never said it is printed out. I am thinking about something about "best practice", eg: avoid sensitive data in not-fully-controlled error string. It is not printed out now, but no one can promise that nobody wouldn't make mistakes in future. Anyway, that's fine to me, just some thoughts about how to make it more stable.